src/main/java/org/sharedid/endpoint/handler/ReadOptOutHandler.java

```diff
@@ -0,0 +1,47 @@
+package org.sharedid.endpoint.handler;
+
+import io.vertx.core.Handler;
+import io.vertx.core.http.Cookie;
+import io.vertx.ext.web.RoutingContext;
+import org.sharedid.endpoint.context.DataContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ReadOptOutHandler implements Handler<RoutingContext> {
+    private String optOutCookieName;
+    private String legacyOptOutCookieName;
```

Comment:
Do we need ReadOptOutHandler? If I am not wrong, opt out is based on logic `cookie('sharedid') == cookie.shared-id.opt-out-value`, it looks like there is no code set cookie `sharedid_optout` value

Reply:
The opt out cookie is set in the "ProcessOptOutHandler," which handles the logic of the "/optout" route.

Reply:
@coryhammon1 I still not understand it. Inside `ProcessOptOutHandler`, I only find code 
```
routingContext.put("opt_out_cookie_name", optOutCookieName);
```
which passes the opt out cookie name to `optout.hbs`. I can't find any code add that cookie.

Reply:
The code that adds the cookie is above that, line 74. Unless I'm misunderstanding your question.

Reply:
There are 2 cookies: `${cookie.shared-id.name}` and `${cookie.opt-out.name}`
line 74 add cookie `${cookie.shared-id.name}` and set its value to `${cookie.shared-id.opt-out-value}`
but there is no code add cookie: `${cookie.opt-out.name}`

Reply:
Get resolved.